### PR TITLE
DRM-17 - Added detail page routing and component with dummy data

### PR DESCRIPTION
**Pull request by Stajgar**
Jira: https://dareit.atlassian.net/browse/DRM-17

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -20,9 +21,9 @@ export const App: React.FC<{}> = () => (
             <Route exact path="/">
               <HomePage />
             </Route>
-            {ITProfessions.map((r) => (
-              <Route exact path={`/${r}`} key={r}>
-                <p>Works {r}</p>
+            {ITProfessions.map((profession) => (
+              <Route exact path={`/details/${profession.name}`} key={profession.name}>
```
Comment (vilanelle) [inline src/App.tsx]: I like it a lot how we have a common sub-route for all professions (`/details`)! 👍

File: src/ITProfessions.ts [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+  },
+  {
+    name: 'Devops',
+    description: 'Devops is awesome',
+  },
+  {
+    name: 'Project-Manager',
+    description: 'Project-Manager is awesome',
+  },
+];
```
Comment (vilanelle) [inline src/ITProfessions.ts]: It's great that we are keeping our state (professions data) in one place, meaning we have a single source of truth for the app! 
I'd also go a step further and added another prop: `route`. At the moment, we are using `name` as path for our routes and that may cause some issues. For example, we have to add a hyphen in `Project-Manager` to be able to use it for route, but we probably don't want the hyphen if use the `name` for page header. 
We could also add `id` prop that would not be subject to change (`name` may easily change and we want to have some constant thing to track professions by). So in the end we could have something like this:
```
{
    id: 'uxUi',
    name: 'UX-UI',
    description: 'UX-UI is awesome',
    route: 'ux-ui',
},
```

File: src/ITProfessions.ts [hunk 3]
```diff
@@ -1,7 +1,22 @@
-export const ITProfessions = [
-  'Backend',
-  'Frontend',
-  'UX-UI',
-  'Devops',
-  'QA',
-  'Project-Manager'];
+export const ITProfessions: { name: string, description: string }[] = [
```
Comment (vilanelle) [inline src/ITProfessions.ts]: It's great that we have type checking here 👏 Maybe we could also extract it to an interface, like so:
```
interface ITProfession {
   name: string;
   description: string;
}
```
and: 
```
export const ITProfessions: ITProfession[] = ...
```
At present, we are inlining it and that's totally fine, but if our interface grows (see my other comment) a separate interface may be more readable :)

**Review by vilanelle — CHANGES_REQUESTED** [review]

File: src/ITProfessions.ts (hunk 2 above)
Comment (vilanelle) [reply src/ITProfessions.ts]: I actually suggested similar changes concerning our state / data to @ireshka in her PR: https://github.com/vilanelle/tech-transition/pull/4#discussion_r571227394. There I described how we would like our state to look like in the end. If you wish, you can apply it here or you can just do the smaller change (as described above) for the scope of this PR, I just wanted to give you a heads up :)

File: src/App.tsx [hunk 4]
```diff
@@ -20,9 +21,9 @@ export const App: React.FC<{}> = () => (
...
+                <DetailPage profession={profession.name} description={profession.description} />
```
Comment (klaraborowska) [inline src/App.tsx]: We will need also other information inside details component (for example icon or color for each path), so I would suggest to pass the whole profession object as props.

File: src/ITProfessions.ts (hunk 3 above)
Comment (Stajgar) [reply src/ITProfessions.ts]: Done

File: src/App.tsx (hunk 4 above)
Comment (Stajgar) [reply src/App.tsx]: Done

File: src/ITProfessions.ts (hunk 2 above)
Comment (Stajgar) [reply src/ITProfessions.ts]: I've change everything as you suggested. As for addressing changes in Gosia's PR, I think the last of us who'll be merging PR should unify the changes.

File: src/App.tsx [hunk 5]
```diff
@@ -20,9 +21,9 @@ export const App: React.FC<{}> = () => (
             <Route exact path="/">
               <HomePage />
             </Route>
-            {ITProfessions.map((r) => (
-              <Route exact path={`/${r}`} key={r}>
-                <p>Works {r}</p>
+            {ITProfessions.map((profession) => (
+              <Route exact path={`/details/${profession.route}`} key={profession.route}>
```
Comment (vilanelle) [inline src/App.tsx]: Typically, we'd use `id` as `key`, because it's should be a unique value - although in this case `route` also meets this requirement.

**Review by vilanelle — APPROVED** [review]

**Review by klaraborowska — APPROVED** [review]

File: src/App.tsx (hunk 5 above)
Comment (Stajgar) [reply src/App.tsx]: Ok, I've changed it

**Pull request merged**